# Setting Everything Up

In [1]:
import pandas as pd
from datetime import datetime

In [2]:
#import the file from the web
import requests
from bs4 import BeautifulSoup
import pandas as pd
from unicodedata import normalize
import re
from datetime import datetime
from pandas import read_csv 
from sklearn.feature_extraction.text import CountVectorizer
import json
from flatten_json import flatten


# I can give a number or use None to remove maximum ceiling & display all columns
pd.options.display.max_columns = None

# # I want to be able to see the entire narrative, so remove the maximum width for each column
# pd.options.display.max_colwidth = None

# pd.options.display.float_format = '{:,.0f}'.format

import string

%matplotlib inline

In [3]:
%load_ext rpy2.ipython
%load_ext autoreload
%autoreload 2

%matplotlib inline  
from matplotlib import rcParams
rcParams['figure.figsize'] = (8, 8)

import warnings
from rpy2.rinterface import RRuntimeWarning
warnings.filterwarnings("ignore") # Ignore all warnings
# warnings.filterwarnings("ignore", category=RRuntimeWarning) # Show some warnings

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from IPython.display import display, HTML

In [4]:
%%javascript
// Disable auto-scrolling
IPython.OutputArea.prototype._should_scroll = function(lines) {
    return false;
}

<IPython.core.display.Javascript object>

In [5]:
%%R

# My commonly used R imports

require('tidyverse')

R[write to console]: Loading required package: tidyverse



── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.4.0      ✔ purrr   1.0.1 
✔ tibble  3.1.8      ✔ dplyr   1.0.10
✔ tidyr   1.3.0      ✔ stringr 1.5.0 
✔ readr   2.1.3      ✔ forcats 1.0.0 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


In [6]:
#To Download PDFs
from urllib.parse import urljoin

In [7]:
#To import camelot and PDF-related items
import camelot
import ghostscript
import sys

In [8]:
%%R

library(jpeg)
library(wordcloud)
library(RColorBrewer)
library(wordcloud2)
library(tm)


R[write to console]: Loading required package: RColorBrewer

R[write to console]: Loading required package: NLP

R[write to console]: 
Attaching package: ‘NLP’


R[write to console]: The following object is masked from ‘package:ggplot2’:

    annotate




In [9]:
from sklearn.feature_extraction.text import CountVectorizer

In [10]:
%%R
library(vistime)
library(shiny)
library(plotly)
library(ggrepel)
#for timeline plotting

R[write to console]: 
Attaching package: ‘plotly’


R[write to console]: The following object is masked from ‘package:ggplot2’:

    last_plot


R[write to console]: The following object is masked from ‘package:stats’:

    filter


R[write to console]: The following object is masked from ‘package:graphics’:

    layout




In [11]:
from collections import Counter


# How do different variables affect education outcomes in Kenya

# Reading the data

In [73]:
df_kenya_education=pd.read_csv("df_edu_no_total_updated.csv")
df_kenya_education

Unnamed: 0.2           County equitable_share_2020_21  \
0              0          Baringo                5,095.65   
1              1            Bomet                5,507.10   
2              2          Bungoma                8,893.65   
3              3            Busia                6,013.50   
4              7  Elgeyo Marakwet                3,861.30   
5              8             Embu                4,304.40   
6              9          Garissa                7,026.30   
7             13         Homa Bay                6,741.45   
8             14           Isiolo                4,241.10   
9             18          Kajiado                6,424.95   
10            19         Kakamega               10,412.85   
11            20          Kericho                5,380.50   
12            21           Kiambu                9,431.70   
13            25           Kilifi               10,444.50   
14            26        Kirinyaga                4,241.10   
15            27            Kisii                7,785.90   
16            28           Kisumu                6,836.40   
17            29            Kitui                8,830.35   
18            30            Kwale                7,785.90   
19            31         Laikipia                4,177.80   
20            32             Lamu                2,595.30   
21            33         Machakos                7,754.25   
22            37          Makueni                7,406.10   
23            41          Mandera               10,222.95   
24            42         Marsabit                6,773.10   
25            43             Meru                8,039.10   
26            44           Migori                6,773.10   
27            45          Mombasa                7,057.95   
28            46         Murang’a                6,298.35   
29            47     Nairobi City               15,919.95   
30            48           Nakuru               10,476.15   
31            49            Nandi                5,348.85   
32            50            Narok                8,039.10   
33            51          Nyamira                4,810.80   
34            52        Nyandarua                4,874.10   
35            53            Nyeri                5,412.15   
36            54          Samburu                4,620.90   
37            58            Siaya                5,791.95   
38            59     Taita/Taveta                4,241.10   
39            60       Tana River                5,855.25   
40            64   Tharaka -Nithi                3,924.60   
41            65      Trans Nzoia                5,760.30   
42            66          Turkana               10,539.45   
43            67      Uasin Gishu                6,330.00   
44            68           Vihiga                4,652.55   
45            72            Wajir                8,545.50   
46            73       West Pokot                5,000.70   

   equitable_share_2021_22_plus conditional_grant_2020_21  \
0                      6,369.39                    752.87   
1                      6,691.10                    832.35   
2                     10,659.44                    876.59   
3                      7,172.16                    845.88   
4                      4,606.53                    735.21   
5                      5,125.24                    998.06   
6                      7,927.21                  1,651.83   
7                      7,805.35                    704.95   
8                      4,710.39                    942.48   
9                      7,954.77                    927.74   
10                    12,389.41                  1,499.98   
11                     6,430.66                    780.03   
12                    11,717.53                  1,418.16   
13                    11,641.59                  1,807.24   
14                     5,196.18                    592.59   
15                     8,894.27                  1,244.00   
16                     8,026.14       

# Reading the data in R

In [74]:
%%R 

# df_kenya_education_r_incorrect_female <- read_csv('df_x_y_cleaned_with_teacher_ratio.csv', show_col_types = FALSE)
df_kenya_education_no_total_r <- read_csv('df_edu_no_total_updated.csv', show_col_types = FALSE)

New names:
• `` -> `...1`


# Background guide for regression analysis

# So far, I am investigating the following y variables:

1) Never been to school
2) Left school before completion
3) NASMLA scores

# So far, I am looking at working the following x variables to create a model to predict the y variables mentioned:
I need to change that: maybe there is a way to do that altogether. Some possible x variables could be:
1) Number of schools per area (how do I normalise by population?)
2) Average school size
3) Pupil Teacher ratio
4) Pupil textbook ratio
5) Electricity
6) Digital Devices
7) County Government revenues
8) Private/ public dichotomy

# For Linear regressions
1) I think I need to go variable by variable. Currently using never been to school as the y variable because that is the measure of access to education I am using. 
2) Given findings on dropout rate, I think it would be an interesting thing to investigate, but not sure whether a sign of outliers would make for a good model?

In [75]:
# model <- lm(Y ~ X, data)
# create a linear regression


# summary(model)
# get a summary (slope, intercept, R2, p-values)


# predict(model) 
# the Y values your model predicts (points on the red line)


# residuals(model)
# Errors...ie, distances from the regression line to each dot)


# rstandard(model)
# Errors, but expressed as a z-score (error expressed as a number of standard deviations from the mean error)


In [76]:
# %%R 
# # guide for linear regression
# model <- lm(Y ~ X, data)
# summary(model)

# County Government Revenues (X1)

#### Dropout rates

In [77]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ total_revenue_2021_22_plus, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    total_revenue_2021_22_plus, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-14.0975  -7.8926  -0.6719   7.0521  18.7073 

Coefficients:
                             Estimate Std. Error t value Pr(>|t|)    
(Intercept)                46.5198369  3.0308114  15.349   <2e-16 ***
total_revenue_2021_22_plus -0.0006439  0.0002696  -2.389   0.0212 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 9.748 on 45 degrees of freedom
Multiple R-squared:  0.1125,	Adjusted R-squared:  0.09279 
F-statistic: 5.705 on 1 and 45 DF,  p-value: 0.02117



# What do we learn from above regression?

This runs dropout rate (Y) against revenue (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable. That is, the revenues affect the dropout rates<br />
The multiple r squared value is 11%, meaning that only 11% of the dropout rate is explained by the revenue of a county

#### Access to education rates

In [78]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ total_revenue_2021_22_plus, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    total_revenue_2021_22_plus, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-15.348 -12.066  -9.659   2.256  57.414 

Coefficients:
                             Estimate Std. Error t value Pr(>|t|)   
(Intercept)                22.3374121  6.3967166   3.492  0.00109 **
total_revenue_2021_22_plus -0.0002344  0.0005690  -0.412  0.68238   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 20.57 on 45 degrees of freedom
Multiple R-squared:  0.003756,	Adjusted R-squared:  -0.01838 
F-statistic: 0.1696 on 1 and 45 DF,  p-value: 0.6824



# What do we learn from above regression?

This runs access to education (Y) against revenue (x). <br />
The p-value is high. Therefore, the x variable does not affect the dependent variable in a statistically significant way. That is, the revenues do not affect access to education in a statistically significant way<br />


#### NASMLA Math Scores

In [107]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ total_revenue_2021_22_plus, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ total_revenue_2021_22_plus, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-99.888 -26.675  -2.308  27.610 131.919 

Coefficients:
                            Estimate Std. Error t value Pr(>|t|)    
(Intercept)                4.693e+02  1.351e+01  34.748   <2e-16 ***
total_revenue_2021_22_plus 3.085e-03  1.201e-03   2.567   0.0136 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 43.44 on 45 degrees of freedom
Multiple R-squared:  0.1278,	Adjusted R-squared:  0.1084 
F-statistic: 6.592 on 1 and 45 DF,  p-value: 0.01364



# What do we learn from above regression?

This runs dropout rate (Y) against revenue (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable. That is, the revenues affect the mathematics scores in a statistically signifcant way<br />
The multiple r squared value is 12%, meaning that only 12% of the variance in math scores can be explained by the revenue of a county

# Average School Size (X2)
From plotting, unlikely to have impact because school size might be too flimsy a variable. For instance, the school could be big but have a lot of teachers and a lot of revenue etc or the school could be small and in that smallness have no teachers or revenue etc. But always still worth exploring

#### Dropout rates

In [79]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ average_school_size, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    average_school_size, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-21.9975  -8.2090   0.5241   7.4610  21.1904 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)         28.88798    6.20813   4.653 2.89e-05 ***
average_school_size  0.03719    0.01997   1.862   0.0691 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 9.97 on 45 degrees of freedom
Multiple R-squared:  0.07154,	Adjusted R-squared:  0.05091 
F-statistic: 3.468 on 1 and 45 DF,  p-value: 0.06912



# What do we learn from above regression?

This runs dropout rate (Y) against average school size (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the average school size does not affect the dropout rates in a statistically significant way<br />
Question: Once p-value is high, is it worth analysing the multiple r-squared? The multiple r squared value is 7%, meaning that only 7% of the variance in dropout rate is explained by the revenue of a county

#### Access to education rates

In [80]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ average_school_size, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    average_school_size, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-18.303 -12.982  -8.769   2.531  54.955 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)  
(Intercept)         33.46571   12.66750   2.642   0.0113 *
average_school_size -0.04452    0.04075  -1.093   0.2804  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 20.34 on 45 degrees of freedom
Multiple R-squared:  0.02584,	Adjusted R-squared:  0.004194 
F-statistic: 1.194 on 1 and 45 DF,  p-value: 0.2804



# What do we learn from above regression?

This runs access to education rate (Y) against average school size (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the average school size does not affect the dropout rates in a statistically significant way<br />
Question: Once p-value is high, is it worth analysing the multiple r-squared? The multiple r squared value is 2%, meaning that only 2% of the variance in access to education is explained by the revenue of a county

#### NASMLA Math Scores

In [108]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ average_school_size, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ average_school_size, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-102.927  -26.523   -9.942   23.064  133.815 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)         507.37913   28.94105  17.531   <2e-16 ***
average_school_size  -0.02452    0.09311  -0.263    0.794    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 46.48 on 45 degrees of freedom
Multiple R-squared:  0.001539,	Adjusted R-squared:  -0.02065 
F-statistic: 0.06934 on 1 and 45 DF,  p-value: 0.7935



# What do we learn from above regression?

This runs NASMLA mathematics scores (Y) against average school size (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the average school size does not affect the math scores in a statistically significant way<br />

# Pupil Teacher Ratio (X3)

#### Dropout rates

In [81]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ overall_pri_ptr_PTR, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    overall_pri_ptr_PTR, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-20.4063  -8.0336  -0.1463   7.9727  17.2160 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          33.6839     5.5259   6.096 2.44e-07 ***
overall_pri_ptr_PTR   0.1980     0.1539   1.287    0.205    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 9.859 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.03629,	Adjusted R-squared:  0.01438 
F-statistic: 1.657 on 1 and 44 DF,  p-value: 0.2048



# What do we learn from above regression?

This runs dropout rate (Y) against the overall primary school pupil teacher ratio (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the overall primary school pupil teacher ratio does not affect the dropout rates in a statistically significant way<br />


#### Access to education rates

In [82]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ overall_pri_ptr_PTR, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    overall_pri_ptr_PTR, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-22.778 -10.748  -3.127   3.530  44.522 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)         -25.3788     9.1516  -2.773  0.00811 ** 
overall_pri_ptr_PTR   1.3195     0.2548   5.178 5.34e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 16.33 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.3787,	Adjusted R-squared:  0.3646 
F-statistic: 26.82 on 1 and 44 DF,  p-value: 5.336e-06



# What do we learn from above regression?

This runs access to education rate (Y) against overall primary school pupil teacher ratio (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, the overall primary school pupil teacher ratio affects the access to education rates in a statistically significant way<br />
The multiple r squared value is 37%, meaning that 37% of the variance in access to education is explained by the overall primary school pupil teacher ratio of a county

#### NASMLA Math Scores

In [109]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ overall_pri_ptr_PTR, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ overall_pri_ptr_PTR, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-82.996 -21.911  -6.994  22.227 101.065 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)         550.3596    22.3328  24.644   <2e-16 ***
overall_pri_ptr_PTR  -1.5370     0.6218  -2.472   0.0174 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 39.84 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.1219,	Adjusted R-squared:  0.102 
F-statistic:  6.11 on 1 and 44 DF,  p-value: 0.01738



# What do we learn from above regression?

This runs NASMLA math scores (Y) against overall primary school pupil teacher ratio (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, the overall primary school pupil teacher ratio affects the math scores in a statistically significant way<br />
The multiple r squared value is 12%, meaning that 12% of the variance in access to education is explained by the overall primary school pupil teacher ratio of a county

# Pupil Textbook Ratio (X4)

#### Dropout rates

In [83]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ pupil_textbook_ratio, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    pupil_textbook_ratio, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-19.2683  -8.8875  -0.5952   8.0360  22.1088 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)             59.64      15.47   3.856 0.000365 ***
pupil_textbook_ratio   -20.13      15.89  -1.267 0.211635    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 10.17 on 45 degrees of freedom
Multiple R-squared:  0.03445,	Adjusted R-squared:  0.01299 
F-statistic: 1.606 on 1 and 45 DF,  p-value: 0.2116



# What do we learn from above regression?

This runs dropout rate (Y) against the pupil textbook ratio (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the pupil textbook ratio does not affect the dropout rates in a statistically significant way<br />


#### Access to education rates

In [84]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ pupil_textbook_ratio, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    pupil_textbook_ratio, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-15.924 -12.057  -8.707   1.926  54.093 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)
(Intercept)            -14.18      30.94  -0.458    0.649
pupil_textbook_ratio    35.29      31.78   1.110    0.273

Residual standard error: 20.34 on 45 degrees of freedom
Multiple R-squared:  0.02666,	Adjusted R-squared:  0.005034 
F-statistic: 1.233 on 1 and 45 DF,  p-value: 0.2728



# What do we learn from above regression?

This runs access to education rate (Y) against the pupil textbook ratio (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the pupil textbook ratio does not affect the access to education rates in a statistically significant way<br />


#### Math scores

In [110]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ pupil_textbook_ratio, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ pupil_textbook_ratio, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-103.624  -24.680   -9.081   23.083  131.966 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)            442.15      70.23   6.296 1.13e-07 ***
pupil_textbook_ratio    59.67      72.15   0.827    0.413    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 46.17 on 45 degrees of freedom
Multiple R-squared:  0.01497,	Adjusted R-squared:  -0.006916 
F-statistic: 0.6841 on 1 and 45 DF,  p-value: 0.4126



# What do we learn from above regression?

This runs mathematics scores (Y) against the pupil textbook ratio (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, the pupil textbook ratio does not affect the math scores in a statistically significant way<br />


# Electricity (X5)

#### Dropout rates

In [85]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ no_power_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    no_power_pct, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-19.3294  -9.0958   0.5388   7.5827  20.6818 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  40.17934    1.82293  22.041   <2e-16 ***
no_power_pct -0.01983    0.38524  -0.051    0.959    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 10.35 on 45 degrees of freedom
Multiple R-squared:  5.888e-05,	Adjusted R-squared:  -0.02216 
F-statistic: 0.00265 on 1 and 45 DF,  p-value: 0.9592



# What do we learn from above regression?

This runs dropout rate (Y) against access to electricity (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, access to electricity does not affect the dropout rates in a statistically significant way<br />


#### Access to Education rates

In [86]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ no_power_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    no_power_pct, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-14.266  -8.533  -4.025  -1.251  59.916 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)   11.3246     2.8029   4.040 0.000206 ***
no_power_pct   3.2732     0.5923   5.526 1.57e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 15.91 on 45 degrees of freedom
Multiple R-squared:  0.4042,	Adjusted R-squared:  0.391 
F-statistic: 30.53 on 1 and 45 DF,  p-value: 1.57e-06



# What do we learn from above regression?

This runs access to education rate (Y) against access to electricity (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, access to electricity affects access to education rates in a statistically significant way<br />
The multiple r-squared value is 40%, meaning 40% of the variance in access to education can be explained by access to electricity


#### Math scores

In [111]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ no_power_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ no_power_pct, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-104.578  -24.391   -9.872   24.169  128.137 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)   505.130      8.079  62.520   <2e-16 ***
no_power_pct   -1.944      1.707  -1.139    0.261    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 45.86 on 45 degrees of freedom
Multiple R-squared:  0.02801,	Adjusted R-squared:  0.006411 
F-statistic: 1.297 on 1 and 45 DF,  p-value: 0.2608



# What do we learn from above regression?

This runs mathematics scores (Y) against access to electricity (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, access to electricity does not affect the math scores in a statistically significant way<br />


# Digital Devices (X6)

#### Dropout rates

In [87]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ pct_installed, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    pct_installed, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-19.5381  -6.5678  -0.0363   7.7892  19.8489 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)
(Intercept)     2.4378    31.1699   0.078    0.938
pct_installed   0.3832     0.3165   1.211    0.232

Residual standard error: 10.18 on 45 degrees of freedom
Multiple R-squared:  0.03154,	Adjusted R-squared:  0.01002 
F-statistic: 1.465 on 1 and 45 DF,  p-value: 0.2324



# What do we learn from above regression?

This runs dropout rate (Y) against access to digital devices (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, access to digital devices does not affect the dropout rates in a statistically significant way<br />


#### Access to education rates

In [88]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ pct_installed, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    pct_installed, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-26.087  -9.825  -7.482   0.443  59.775 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)   
(Intercept)   175.2730    58.6853   2.987  0.00455 **
pct_installed  -1.5785     0.5959  -2.649  0.01110 * 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 19.17 on 45 degrees of freedom
Multiple R-squared:  0.1349,	Adjusted R-squared:  0.1156 
F-statistic: 7.016 on 1 and 45 DF,  p-value: 0.0111



# What do we learn from above regression?

This runs access to education rate (Y) against access to a complete set of digital devices (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, access to a complete set of digital devicess affects access to education rates in a statistically significant way<br />
The multiple r-squared value is 13%, meaning 13% of the variance in access to education can be explained by access to electricity


#### Math scores

In [112]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ pct_installed, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ pct_installed, data = df_kenya_education_no_total_r)

Residuals:
   Min     1Q Median     3Q    Max 
-96.61 -27.85 -11.66  26.22 131.28 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)   
(Intercept)    386.977    141.382   2.737  0.00884 **
pct_installed    1.149      1.436   0.800  0.42785   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 46.19 on 45 degrees of freedom
Multiple R-squared:  0.01403,	Adjusted R-squared:  -0.007884 
F-statistic: 0.6402 on 1 and 45 DF,  p-value: 0.4278



# What do we learn from above regression?

This runs math scores (Y) against access to digital devices (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, access to digital devices does not affect the math scores in a statistically significant way<br />


# Private Public Dichotomy (X7)

#### Dropout rates

In [89]:
%%R 
# guide for linear regression
model <- lm(left_school_over_left_before_and_after_as_pct ~ public_over_overall_enrolment_as_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    public_over_overall_enrolment_as_pct, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-17.5851  -5.7137   0.3488   7.5389  15.8606 

Coefficients:
                                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)                           -7.3613    10.6351  -0.692    0.492    
public_over_overall_enrolment_as_pct   0.5641     0.1254   4.497 4.81e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 8.595 on 45 degrees of freedom
Multiple R-squared:   0.31,	Adjusted R-squared:  0.2947 
F-statistic: 20.22 on 1 and 45 DF,  p-value: 4.813e-05



# What do we learn from above regression?

This runs dropout rate (Y) against the rate of enrolment in public school (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, enrolment in public school vs private school affects dropout rates in a statistically significant way<br />
The multiple r-squared value is 31%, meaning 31% of the variance in dropout rates can be explained by rates of enrolment in public school


#### Access to education rates

In [90]:
%%R 
# guide for linear regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ public_over_overall_enrolment_as_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    public_over_overall_enrolment_as_pct, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-14.563 -11.825  -9.915   1.582  57.193 

Coefficients:
                                     Estimate Std. Error t value Pr(>|t|)
(Intercept)                           10.2026    25.4621   0.401    0.691
public_over_overall_enrolment_as_pct   0.1165     0.3003   0.388    0.700

Residual standard error: 20.58 on 45 degrees of freedom
Multiple R-squared:  0.003333,	Adjusted R-squared:  -0.01882 
F-statistic: 0.1505 on 1 and 45 DF,  p-value: 0.6999



# What do we learn from above regression?

This runs access to education rate (Y) against public school enrolment rates (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, public school enrolment ratesdoes not affect the access to education rates in a statistically significant way<br />


#### Math scores

In [113]:
%%R 
# guide for linear regression
model <- lm(mathematics_mean ~ public_over_overall_enrolment_as_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ public_over_overall_enrolment_as_pct, 
    data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-135.260  -24.483   -0.991   23.612  119.917 

Coefficients:
                                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)                          663.9187    52.0287  12.761  < 2e-16 ***
public_over_overall_enrolment_as_pct  -1.9474     0.6137  -3.173  0.00272 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 42.05 on 45 degrees of freedom
Multiple R-squared:  0.1829,	Adjusted R-squared:  0.1647 
F-statistic: 10.07 on 1 and 45 DF,  p-value: 0.002717



# What do we learn from above regression?

This runs math scores (Y) against the rate of enrolment in public school (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, enrolment in public school vs private school affects math scores in a statistically significant way<br />
The multiple r-squared value is 18%, meaning 18% of the variance in math scores can be explained by rates of enrolment in public school


# The question of school density

How do I check for whether distance to school affects education outcomes?<br />
There is no straightforward way in this moment. <br />
I could work with number of schools over area. <br />
But on a practical level that would not be properly accurate since some areas are less dense. <br />
So I need a parameter that I: <br />
    1) Should consider an estimate <br />
    2) Should count in number of schools, area and population of the county: <br />
   > Population per number of schools will give me number of people served per school. If I divide this by area, I can get number of people served per school per area. It is not perfect but it could be something to work with. <br />
 Another question: Wouldn't a multiple regression do this on my behalf without my input? 

# School Density (X8)

In [104]:
#### Dropout rates

# %%R 
# # guide for linear regression
# model <- lm(left_school_over_left_before_and_after_as_pct ~ population_per_no_of_schools_over_area, df_kenya_education_no_total_r)
# summary(model)

In [101]:
%%R 
# guide for linear regression
school_density_variable<-(df_kenya_education_no_total_r$population/df_kenya_education_no_total_r$number_of_schools)/df_kenya_education_no_total_r$land_area_km_sq
model <- lm(left_school_over_left_before_and_after_as_pct ~ school_density_variable, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    school_density_variable, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-20.821  -8.308  -0.082   8.377  22.102 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)    
(Intercept)               37.922      2.285   16.60   <2e-16 ***
school_density_variable    5.865      4.846    1.21    0.233    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 10.25 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.03222,	Adjusted R-squared:  0.01022 
F-statistic: 1.465 on 1 and 44 DF,  p-value: 0.2326



# What do we learn from above regression?

This runs dropout rate (Y) against a variable of school density (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, school density does not affect the dropout rates in a statistically significant way<br />


#### Access to education rates

In [100]:
%%R 
# guide for linear regression
school_density_variable<-(df_kenya_education_no_total_r$population/df_kenya_education_no_total_r$number_of_schools)/df_kenya_education_no_total_r$land_area_km_sq

model <- lm(never_been_to_school_or_learning_institution_pct ~ school_density_variable, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    school_density_variable, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-21.081 -11.334  -5.512   8.424  46.259 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)    
(Intercept)               32.505      3.919   8.293 1.54e-10 ***
school_density_variable  -34.779      8.313  -4.184 0.000135 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 17.59 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.2846,	Adjusted R-squared:  0.2683 
F-statistic:  17.5 on 1 and 44 DF,  p-value: 0.0001348



# What do we learn from above regression?

This runs access to education rates (Y) against a variable of school density normalised over population, area and number of schools (x). <br />
The p-value is low. Therefore, we can conclude that the explanatory variable affects the dependent variable in a statistically significant way. That is, school density affects access to education rates in a statistically significant way<br />
The multiple r-squared value is 28%, meaning 28% of the variance in access to education rates can be explained by rates of enrolment in public school


In [ ]:
#### Math scores

In [194]:
%%R 
# guide for linear regression
school_density_variable<-(df_kenya_education_no_total_r$population/df_kenya_education_no_total_r$number_of_schools)/df_kenya_education_no_total_r$land_area_km_sq

model <- lm(mathematics_mean ~ school_density_variable, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ school_density_variable, data = df_kenya_education_no_total_r)

Residuals:
    Min      1Q  Median      3Q     Max 
-95.900 -25.020  -9.137  24.088 127.208 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)    
(Intercept)               492.95      10.37  47.518   <2e-16 ***
school_density_variable    18.59      22.00   0.845    0.403    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 46.56 on 44 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.01597,	Adjusted R-squared:  -0.006395 
F-statistic: 0.7141 on 1 and 44 DF,  p-value: 0.4027



# What do we learn from above regression?

This runs math scores (Y) against a variable of school density (x). <br />
The p-value is high. Therefore, we can conclude that the explanatory variable does not affect the dependent variable in a statistically significant way. That is, school density does not affect the mathematics scores in a statistically significant way<br />


# For multivariable regressions:
1) For dropout rates (left_school_over_left_before_and_after_as_pct)
2) For access to education rates (never_been_to_school_or_learning_institution_pct)
3) For math scores (mathematics_mean)

#### Here is a list of the X variables titles and column names for reference:
1) County Government revenues (total_revenue_2021_22_plus)
2) Average school size (average_school_size)
3) Pupil Teacher ratio (overall_pri_ptr_PTR)
4) Pupil textbook ratio (pupil_textbook_ratio)
5) Electricity (no_power_pct)
6) Digital Devices (pct_installed)
7) Private/ public dichotomy (public_over_overall_enrolment_as_pct)
8) School density normalised by population, number of schools, and area (school_density_variable)

## For Dropout rates

In [115]:
# ### Every variable
# %%R 
# # guide for multivariable regression
# model <- lm(left_school_over_left_before_and_after_as_pct ~ total_revenue_2021_22_plus + average_school_size + overall_pri_ptr_PTR + pupil_textbook_ratio + no_power_pct + pct_installed + public_over_overall_enrolment_as_pct + school_density_variable, df_kenya_education_no_total_r)
# summary(model)

In [193]:
%%R 
# guide for multivariable regression
model <- lm(left_school_over_left_before_and_after_as_pct ~total_revenue_2021_22_plus + average_school_size+ pupil_textbook_ratio  + public_over_overall_enrolment_as_pct+ school_density_variable, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = left_school_over_left_before_and_after_as_pct ~ 
    total_revenue_2021_22_plus + average_school_size + pupil_textbook_ratio + 
        public_over_overall_enrolment_as_pct + school_density_variable, 
    data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-15.4850  -5.5526   0.1198   5.3399  18.5502 

Coefficients:
                                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)                           3.422e-01  1.764e+01   0.019 0.984625    
total_revenue_2021_22_plus           -4.106e-04  2.932e-04  -1.401 0.169060    
average_school_size                   3.834e-02  2.098e-02   1.827 0.075158 .  
pupil_textbook_ratio                 -1.349e+01  1.245e+01  -1.084 0.285044    
public_over_overall_enrolment_as_pct  5.135e-01  1.438e-01   3.572 0.000941 ***
school_density_variable               5.437e+00  4.486e+00   1.212 0.232667    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0

### For dropout rates, 42% was nearly the highest adjusted r-squared value I could get above. The p values are however quite high

# For Access to Education Rates

In [117]:
# ### Every variable
# %%R 
# # guide for multivariable regression
# model <- lm(never_been_to_school_or_learning_institution_pct ~ total_revenue_2021_22_plus + average_school_size + overall_pri_ptr_PTR + pupil_textbook_ratio + no_power_pct + pct_installed + public_over_overall_enrolment_as_pct + school_density_variable, df_kenya_education_no_total_r)
# summary(model)

In [191]:
%%R 
# guide for multivariable regression
model <- lm(never_been_to_school_or_learning_institution_pct ~ total_revenue_2021_22_plus + average_school_size + overall_pri_ptr_PTR + pct_installed + public_over_overall_enrolment_as_pct + school_density_variable + with_national_grid_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    total_revenue_2021_22_plus + average_school_size + overall_pri_ptr_PTR + 
        pct_installed + public_over_overall_enrolment_as_pct + 
        school_density_variable + with_national_grid_pct, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-19.3877  -2.6088  -0.4802   2.6778  25.4374 

Coefficients:
                                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)                          79.1896421 31.1164260   2.545 0.015232 *  
total_revenue_2021_22_plus           -0.0008881  0.0006918  -1.284 0.207201    
average_school_size                  -0.0513338  0.0354854  -1.447 0.156423    
overall_pri_ptr_PTR                   1.0640865  0.3119358   3.411 0.001578 ** 
pct_installed                        -0.1041078  0.3492883  -0.298 0.767327    
public_over_overall_enrolment_as_pct -0.3846339  0.1770304  -2.173 0.036281 *  
school_dens

In [222]:
%%R 
# guide for multivariable regression
model <- lm(never_been_to_school_or_learning_institution_pct ~total_revenue_2021_22_plus + log(average_school_size) + log(overall_pri_ptr_PTR) + public_over_overall_enrolment_as_pct + with_national_grid_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = never_been_to_school_or_learning_institution_pct ~ 
    total_revenue_2021_22_plus + log(average_school_size) + log(overall_pri_ptr_PTR) + 
        public_over_overall_enrolment_as_pct + with_national_grid_pct, 
    data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-20.0151  -2.0431   0.1631   3.3804  22.0823 

Coefficients:
                                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)                           7.097e+01  3.514e+01   2.020 0.050143 .  
total_revenue_2021_22_plus           -8.507e-04  5.623e-04  -1.513 0.138167    
log(average_school_size)             -2.508e+01  1.075e+01  -2.334 0.024706 *  
log(overall_pri_ptr_PTR)              4.559e+01  1.223e+01   3.727 0.000598 ***
public_over_overall_enrolment_as_pct -3.753e-01  1.517e-01  -2.474 0.017710 *  
with_national_grid_pct               -3.736e-01  7.490e-02  -4.989 1.23e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’

### For access to education rates, 82% was the highest adjusted r-squared value I could get above. 
The p values are all low except for revenue. <br />
Even during the one by one linear regressions, access to education rates also had teh lowest p values. <br />

# For Mathematics Scores

In [223]:
# ### Every variable
# %%R 
# # guide for multivariable regression
# model <- lm(mathematics_mean ~ total_revenue_2021_22_plus + average_school_size + overall_pri_ptr_PTR + pupil_textbook_ratio + no_power_pct + pct_installed + public_over_overall_enrolment_as_pct + school_density_variable, df_kenya_education_no_total_r)
# summary(model)

In [233]:
%%R 
# guide for multivariable regression
model <- lm(mathematics_mean ~ total_revenue_2021_22_plus + public_over_overall_enrolment_as_pct, df_kenya_education_no_total_r)
summary(model)


Call:
lm(formula = mathematics_mean ~ total_revenue_2021_22_plus + 
    public_over_overall_enrolment_as_pct, data = df_kenya_education_no_total_r)

Residuals:
     Min       1Q   Median       3Q      Max 
-126.818  -25.311   -2.709   28.732  129.060 

Coefficients:
                                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)                          610.507159  65.276220   9.353 5.05e-12 ***
total_revenue_2021_22_plus             0.001741   0.001304   1.335   0.1887    
public_over_overall_enrolment_as_pct  -1.518272   0.688095  -2.206   0.0326 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 41.69 on 44 degrees of freedom
Multiple R-squared:  0.2147,	Adjusted R-squared:  0.179 
F-statistic: 6.014 on 2 and 44 DF,  p-value: 0.004911



### For math scores, 17% was the highest adjusted r-squared value I could get above. 
This signifies that conceptualising a different model with different variables would be ideal for predicting math scores